util/CosmosHelper.js

```diff
@@ -122,66 +134,84 @@ export async function getTransferInfo(txHash, opt) {
 }
 
 export async function queryLikeCoinBalance(addr) {
-  if (!api) await initCosmos();
-  const account = await api.get.account(addr);
-  const [amount] = account.coins.filter(coin => coin.denom === COSMOS_DENOM);
+  if (!cosmosClient) await initCosmosClient();
+  const account = await cosmosClient.getAccount(addr);
+  const [amount] = account.balance.filter(balance => balance.denom === COSMOS_DENOM);
   if (!amount) return 0;
   return amountToLIKE(amount);
 }
 
-export async function sendTx(msgCallPromise, signer, {
-  gasPrices = DEFAULT_GAS_PRICE,
-  memo,
-  simulate: isSimulate,
-} = {}) {
-  const { simulate, send } = await msgCallPromise;
-  const gas = (await simulate({ memo })).toString();
-  if (isSimulate) return { gas, gasPrices };
-  const { hash, included } = await send({ gas, gasPrices, memo }, signer);
-  return {
-    txHash: hash,
-    included,
-    gas,
-    gasPrices,
-  };
-}
-
 export async function transfer({
   from,
   to,
   value,
   memo,
-}, signer, { simulate = false } = {}) {
-  if (!api) await initCosmos();
+}, signer) {
   const amount = LIKEToAmount(value);
-  const msgPromise = api.MsgSend(from, { toAddress: to, amounts: [amount] });
-  return sendTx(msgPromise, signer, { memo, simulate });
+  const fee = {
+    amount: [{
+      denom: 'nanolike',
+      amount: '1000',
+    }],
+    gas: '44000',
+  };
+  const defaultGasPrice = GasPrice.fromString('1000nanolike');
```

Comment:
Why do we define both `fee` and `defaultGasPrice`? Which one is really used?

Reply:
defaultGasPrice is used due to signingCosmosClient need to be in GasPrice form. 
However, to not repeat, I will extract fee value for defaultGasPrice.